prime-router/docs/proposals/0020-universal-metadata-search/example-queries.md

```diff
@@ -0,0 +1,196 @@
+    ```sql
+    select * from delivered_reports_for_receiver
+    order by item_count;
+    ```
+  - I want to gain insight into the data I receive by querying the metadata associated with items. See
+    [All facilities & providers](https://www.figma.com/proto/6mwI5ac6rprACKDzDo4Ady/ReportStream-Workspace-%7C-2023?node-id=995%3A13474&scaling=min-zoom&page-id=496%3A6448&starting-point-node-id=995%3A13227&show-proto-sidebar=1).
+      - I want a list of all performing facilities sending data to me.
+   
+      ???
+      - I want a list of all ordering providers sending data to me.
+  
+      ???
+      - I want a list of all submitters sending data to me.
+      ```sql
+      select
+      distinct submitted.sending_org
+      from report_file delivered
+      join terminal_report_ids on terminal_report_id = delivered.report_id
+      join report_file submitted on submitted.report_id = terminal_report_ids.original_report_id
+      where delivered.transport_result is not null
+      and submitted.sending_org = %receiving_org%
+      ```
+      - I want to know the last time a particular performing facility sent data to me.
+    
+      ??? Likely needs to have the facility index to know which receivers got it  
+
+      - I want to know the last time a particular ordering provider sent data to me.
+
+      ??? Likely needs to have the provider index to know which receivers got it
+  
+      - I want to know the last time a particular submitter sent data to me.
+      ```sql
+      select
+      max(submitted.created_at)
+      from report_file delivered
+      join terminal_report_ids on terminal_report_id = delivered.report_id
+      join report_file submitted on submitted.report_id = terminal_report_ids.original_report_id
+      where delivered.transport_result is not null
+      and submitted.sending_org = %receiving_org%
+      ```
+- Given some report item data, I want to find other data associated with the item. See
+  [Carroll Schultz](https://www.figma.com/proto/6mwI5ac6rprACKDzDo4Ady/ReportStream-Workspace-%7C-2023?node-id=1081%3A15935&scaling=min-zoom&page-id=496%3A6448&starting-point-node-id=995%3A13227&show-proto-sidebar=1).
+    - I want to see when an ordering provider first reported to us, the receiver.
+        - Search the ordering provider index and get all the metadata ids
+        ```sql
+        select min(metadata.created_at) from metadata
+        where id in %metadata_ids%
+        join terminal_report_ids on(
+            terminal_report_ids.origin_report_id = metadata.report_id
+            and terminal_report_ids.origin_report_index = metadata.report_index
+        )
+        join report_file on(
+            terminal_report_ids.terminal_report_id = report_file.report_id
+        )
+       where report_file.receiving_org = %receiving_org%
+        ```
+    - I want the average number of tests per report sent to us for all reports including a particular ordering provider.
```

Comment:
Did you mean to skip this and the next one?

Reply:
Yep, ultimately I might not include this unless the team thought it was helpful, but I wanted in the review to hopefully provide some context.

My goal here was to evaluate a subset of the search requirements to see if it we could realistically put the metadata in elasticsearch and continue to store the report/item lineage in the DB.  These questions all suffer from the same flaw of having to grab a huge number of metadata ids out of elastic search and then filtering them down via the DB.